storage-proofs/core/src/api_version.rs

```diff
@@ -1,5 +1,21 @@
+use anyhow::{self, Result};
+use std::str::FromStr;
+
 #[derive(Debug, Copy, Clone)]
 pub enum APIVersion {
     V1_0,
     V1_1,
 }
+
+impl FromStr for APIVersion {
+    type Err = anyhow::Error;
+    fn from_str(api_version: &str) -> Result<Self> {
```

Comment:
Lets associate a `Semver` version with each of the enum and use that for parsing using: https://docs.rs/semver/

Reply:
For this, do we want `ApiVersion::{V1_0_0, V1_1_0}` for the constants also?

Reply:
good question, I am not sure tbh